pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go

```diff
@@ -12,56 +12,63 @@ package tests
 
 import (
 	"context"
+	"math/rand"
 
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
-	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/clusterupgrade"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/mixedversion"
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
-	"github.com/cockroachdb/cockroach/pkg/testutils/release"
-	"github.com/stretchr/testify/require"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
 )
 
-// testSetupResetStep setups the testing state (e.g. create a few databases and tables)
-// and always use declarative schema changer on all nodes.
-// Queries run in this function should be idempotent.
-func testSetupResetStep(c cluster.Cluster) versionStep {
-	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-		db := c.Conn(ctx, t.L(), 1)
-		setUpQuery := `
-CREATE DATABASE IF NOT EXISTS testdb;
-CREATE SCHEMA IF NOT EXISTS testdb.testsc;
-CREATE TABLE IF NOT EXISTS testdb.testsc.t (i INT PRIMARY KEY, j INT NOT NULL, INDEX idx (j), CONSTRAINT check_j CHECK (j > 0));
-INSERT INTO testdb.testsc.t VALUES (1, 1);
-CREATE TABLE IF NOT EXISTS testdb.testsc.t2 (i INT NOT NULL, j INT NOT NULL);
-INSERT INTO testdb.testsc.t2 VALUES (1, 1);
-CREATE TYPE IF NOT EXISTS testdb.testsc.typ AS ENUM ('a', 'b');
-CREATE SEQUENCE IF NOT EXISTS testdb.testsc.s;
-CREATE VIEW IF NOT EXISTS testdb.testsc.v AS (SELECT i*2 FROM testdb.testsc.t);
-`
-		_, err := db.ExecContext(ctx, setUpQuery)
-		require.NoError(t, err)
+func registerDeclarativeSchemaChangerJobCompatibilityInMixedVersion(r registry.Registry) {
+	// declarative_schema_changer/job-compatibility-mixed-version tests that,
+	// in a mixed version cluster, jobs created by the declarative schema changer
+	// are both backward and forward compatible. That is, declarative schema
+	// changer job created by nodes running newer (resp. older) binary versions
+	// can be adopted and finished by nodes running older (resp. newer) binary versions.
+	// This test requires us to come back and change the to-be-tests stmts to be those
+	// supported in the "previous" major release.
+	r.Add(registry.TestSpec{
+		Name:             "declarative_schema_changer/job-compatibility-mixed-version-V231-V232",
```

Comment:
I think in commit 1, this should be `V222-V231`?

Reply:
yes, done - ty